app/controllers/posts_controller.rb

```diff
@@ -85,11 +85,15 @@ def up_vote
   end
 
   def post_params
-    params.require(:post).permit(
-      :body,
-      :kind,
-      {attachments: []}
-    )
+    if attachments_enabled?
+      params.require(:post).permit(
+        :body,
+        :kind,
+        {attachments: []}
+      )
+    else
+      params.require(:post).permit(:body, :kind)
+    end
```

Comment:
This check isn't really necessary, you check in `create` whether `attachments_enabled?` anyways.